components/econet/econet.cpp

```diff
@@ -431,9 +426,15 @@ void Econet::request_strings_() {
     objects.push_back(datapoint_ids_for_read_service_.front());
     datapoint_ids_for_read_service_.pop();
   } else {
-    uint8_t request_mod = read_requests_++ % request_mods_;
-    std::copy(request_datapoint_ids_[request_mod].begin(), request_datapoint_ids_[request_mod].end(),
-              back_inserter(objects));
+    for (int j = 0; j < request_mods_; j++) {
+      uint8_t request_mod = read_requests_++ % request_mods_;
```

Comment:
Is this needed? Can't you just use the j ? I think you can remove this line and rename j with request_mod.
Then you can delete read_requests_.
Can you also delete request_mods_? I think you can iterate up to 8 which btw should be made a constant used here and in the header file.